src/Sentry/Internal/Hub.cs

```diff
@@ -66,6 +66,8 @@ internal Hub(
             PushScope();
         }
 
+        Logger = new DefaultSentryStructuredLogger(this, ScopeManager, options, _clock);
```

Comment:
If `options.Experimental.EnableLogs == false` we could just instantiate a `DisabledSentryStructuredLogger` here.

That way `SentryStructuredLogger.CaptureLog` would be a NoOp rather than [evaluating this](https://github.com/getsentry/sentry-dotnet/blob/6bd4c96ad66c69360e7042c4307ce6f9d3de2d1f/src/Sentry/Internal/DefaultSentryStructuredLogger.cs#L28-L31) every time it's called. And we could even remove that if enabled check from the Default logger since, implicitly, if the logger had been instantiated that means that logging is enabled (could maybe change it to a `Debug.Assert` if we really wanted).

Reply:
- moved creation of `SentryStructuredLogger` to factory method
  - create "enabled" logger when options are enabled
  - create "disabled" looger when options are disabled
- remove all "enabled-checks" from the Default-Logger
  - `Debug.Assert` in the `DefaultSentryStructuredLogger` to developer-assert that the options are meant to be enabled when it's `.ctor` is invoked
- also renamed from `DisabledSentryStructuredLogger` to `NoOpSentryStructuredLogger`
  - to match `NoOpSpan`
  - to match `NoOpTransaction`